### PR TITLE
[bugfix](memory) merge block may allocate failed

### PR DESCRIPTION
**Pull request by yiguolei**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/scan/scanner_scheduler.cpp [hunk 1]
```diff
@@ -350,7 +350,10 @@ void ScannerScheduler::_scanner_scan(ScannerScheduler* scheduler, ScannerContext
             ctx->return_free_block(std::move(block));
         } else {
             if (!blocks.empty() && blocks.back()->rows() + block->rows() <= state->batch_size()) {
-                vectorized::MutableBlock(blocks.back().get()).merge(*block);
+                status = vectorized::MutableBlock(blocks.back().get()).merge(*block);
+                if (!status.ok()) {
+                    break;
```
[the remark this answers is not in the document]
Comment (yiguolei) [reply be/src/vec/exec/scan/scanner_scheduler.cpp:355]: if the status ! = ok, the scanner will set the error status into scanner ctx and set eof=true and cancel the query.

**Comment by yiguolei** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Comment by yiguolei** [comment]
run ut

**Comment by yiguolei** [comment]
run clickbench

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.71 seconds
 stream load tsv:          427 seconds loaded 74807831229 Bytes, about 167 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          60 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230511012114_clickbench_pr_141959.html

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**